### DISCUSSION TITLE
Error attribute oob ip

### DISCUSSION THREAD

**Discussion by max0013** [netbox-community/netbox#14254] [Help Wanted!]
Hello, 

I try to a attribute an IP address to the oob_ip field of a device via API.
When I do it I have an error : 

`The specified IP address (xxx.xxx.xxx.xxx/xx) is not assigned to this device.`

The concerned IP address is affected to a module, which is affected to a module bay attach the the device.
When I try to do the same thing in the GUI it's working like a charm, so, how can I do it through an API call ? 

Thanks for advance

**Comment by candlerb** [comment]
I haven't tested this, but I expect it works in the same way as primary_ip(4|6). If so, this means you first need to create an OOB interface (with mgmt_only=True); then add the IP address to that interface; *then* you can set the oob address on the device itself.

**Reply by max0013** [reply]
Maybe it's not the good way to do it but in my context I have the interface assigned to a module type. 
When I create the module, based on this module_type, the interface was assign, after I assign the ip address to this interface, and after again I try to assign the ip address to the oob_ip field.
I'm prety sure the "mgmt only" check box is not checked, maybe this is the root cause of the error...

**Reply by max0013** [reply]
I just try to check the "management o ly" checkbox on the interface of the module type and retry to affect the address to oob_ip but same result...

**Reply by candlerb** [reply]
I have checked the [code](https://github.com/netbox-community/netbox/blob/v3.6.5/netbox/dcim/models/devices.py#L945-L953): it doesn't matter whether mgmt_only is set or not. It uses exactly the same logic as primary_ip4 and primary_ip6.

You'll need to give the exact steps to reproduce the problem, then it can be reproduced and if necessary turned into a bug report. You'll also need to say what version of Netbox you're using.

**Reply by max0013** [reply]
I use netbox v3.6.5

The steps are the following:
I have a device with a module bay attached on it witch has a module called "idrac" attached to. This module "own" the idrac interface.
Here the interface settings : 

![image](https://github.com/netbox-community/netbox/assets/53040095/49afd7af-0e34-4845-8bd4-d61fa47cdb82)

As you can see, the interface is linked to the device and the module.
In the interfaces section of the device I find the idrac interface and I can see his affected ip address.

![image](https://github.com/netbox-community/netbox/assets/53040095/4b7eaf85-b572-416c-bbdb-46d1729fd1db)

Basically I want to affect this ip address to the Out-of-band IP. If I try to do it with GUI, it work.

![image](https://github.com/netbox-community/netbox/assets/53040095/7e1cd05a-6c80-4eea-9896-f0be65a85861)

If I try to do the same thing with ansible I receive the error.
Here my code : 

`- name: set oob ip
      ansible.builtin.uri:
        url: "{{ netbox_url }}/api/dcim/devices/"
        method: POST
        body_format: json
        status_code: 201
        return_content: true
        headers:
          accept: application/json
          Content-Type: application/json
          Authorization: "Token {{ netbox_token }}"
        body: {
          "device": "{{ netbox_set_device_result.device.id }}",
          "oob_ip": "{{ query('netbox.netbox.nb_lookup', 'ip-addresses',
                api_endpoint=netbox_url,
                api_filter='address=xxx.xxx.xxx.xxx/xx',
                token=netbox_token) | map(attribute='value') | map(attribute='id') | join}} ",
          "device_type": "{{ query('netbox.netbox.nb_lookup', 'device-types',
                api_endpoint=netbox_url,
                api_filter='slug='~slug,
                token=netbox_token) | map(attribute='value') | map(attribute='id') | join}} ",         
          "role": "{{ role_id }}",
          "site": "{{ site | map(attribute='id') | join }}"
        }`

Hope I was clear with y explanation ^^
